roles/openshift_master/handlers/main.yml

```diff
@@ -1,4 +1,4 @@
 ---
 - name: restart master
   service: name={{ openshift.common.service_type }}-master state=restarted
-  when: not openshift_master_ha | bool
+  when: not (openshift_master_ha | bool or skip_master_restart | default(false))
```

Comment:
yes, these handlers can be triggered from other roles that have dependencies on the role they are defined in.

In this case, the fluentd roles have dependencies on the openshfit_{master,node} roles and have tasks which  notify the service restart handlers for those roles...  without the default(false) filter, they cause the fluentd roles to fail (even when they are fully skipped otherwise).